File: 0455-assign-cookies/0455-assign-cookies.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort()
        s.sort()
        ans = 0
        pt = 0
        for i in g:
            while pt < len(s) and s[pt] < i:
                pt += 1
            if pt == len(s):
                return ans
            ans += 1
            pt += 1
        return ans
            

File: 0455-assign-cookies/0455-assign-cookies.go
func findContentChildren(g []int, s []int) int {
    sort.Ints(g)
    sort.Ints(s)
    ans := 0
    pt := len(s) - 1 
    for idx := len(g) - 1; idx >= 0; idx-- {
        if pt >= 0 && s[pt] >= g[idx] {
            ans++
            pt--
        }
    }
    return ans
}